engines/order_management/spec/services/order_management/subscriptions/proxy_order_syncer_spec.rb

```diff
@@ -16,107 +18,131 @@ module Subscriptions
         let(:syncer) { ProxyOrderSyncer.new(subscription) }
 
         context "when the subscription is not persisted" do
-          before do
-            oc # Ensure oc is created before we attempt to sync
-            expect{ syncer.sync! }.to_not change(ProxyOrder, :count).from(0)
+          let(:schedule) { create(:schedule, order_cycles: [oc]) }
+          let(:subscription) do
+            build(
+              :subscription,
+              begins_at: now + 1.minute,
+              ends_at: now + 2.minutes,
+              schedule: schedule
+            )
           end
 
           context "and the schedule includes a closed oc (ie. closed before opens_at)" do
-            let(:oc) { closed_oc }
+            let!(:oc) { closed_oc }
+
             it "does not create a new proxy order for that oc" do
               expect{ subscription.save! }.to_not change(ProxyOrder, :count).from(0)
               expect(order_cycles).to_not include oc
             end
           end
 
           context "and the schedule includes an open oc that closes before begins_at" do
-            let(:oc) { open_oc_closes_before_begins_at_oc }
+            let!(:oc) { open_oc_closes_before_begins_at_oc }
+
             it "does not create a new proxy order for that oc" do
               expect{ subscription.save! }.to_not change(ProxyOrder, :count).from(0)
               expect(order_cycles).to_not include oc
             end
           end
 
           context "and the schedule has an open OC that closes between begins_at and ends_at" do
-            let(:oc) { open_oc }
+            let!(:oc) { open_oc }
+
             it "creates a new proxy order for that oc" do
+              syncer.sync!
+
               expect{ subscription.save! }.to change(ProxyOrder, :count).from(0).to(1)
-              expect(order_cycles).to include oc
+              expect(subscription.reload.proxy_orders.map(&:order_cycle)).to include oc
             end
           end
 
           context "and the schedule includes upcoming oc that closes before begins_at" do
-            let(:oc) { upcoming_closes_before_begins_at_oc }
+            let!(:oc) { upcoming_closes_before_begins_at_oc }
+
             it "does not create a new proxy order for that oc" do
               expect{ subscription.save! }.to_not change(ProxyOrder, :count).from(0)
               expect(order_cycles).to_not include oc
             end
           end
 
           context "and the schedule includes upcoming oc that closes on begins_at" do
-            let(:oc) { upcoming_closes_on_begins_at_oc }
+            let!(:oc) { upcoming_closes_on_begins_at_oc }
+
+            let(:schedule) { create(:schedule, order_cycles: [oc]) }
+            let(:subscription) do
+              build(
+                :subscription,
+                begins_at: now + 1.minute,
+                ends_at: now + 2.minutes,
+                schedule: schedule
+              )
+            end
```

Comment:
I wonder why this is needed if it's already in line 53 above.

Reply:
It can probably be DRYed a bit but it comes at a cost in tests. There were so many tangled dependencies that I had to duplicate all `lets` to remove all dependencies and then fix test by test. In summary, it was DRY but at the expense of changeability.